akka-remote/src/main/scala/akka/remote/Endpoint.scala

```diff
@@ -1035,13 +1035,12 @@ private[remote] class EndpointReader(
       replyTo ! StoppedReading(writer)
 
     case InboundPayload(p) if p.size <= transport.maximumPayloadBytes ⇒
-      val (ackOption, msgOption) = tryDecodeMessageAndAck(p)
-      for (ack ← ackOption; reliableDelivery ← reliableDeliverySupervisor) reliableDelivery ! ack
```

Comment:
I just realized that this is probably not what we thought it was.
I guess we thought this was about sending an Ack for an incoming system message, I think it's the opposite.

This node sent a system message previously and the other side has received it and sending back the ack for that. This is about updating our records about that it has been acked and doesn't have to be resent. In other words is about a previously outgoing system message, not related to a possible incoming system message.

I think we should close this PR since we can't prove that it is fixing anything.

Reply:
:+1: